platforms/core-runtime/logging/src/main/java/org/gradle/internal/featurelifecycle/LoggingDeprecatedFeatureHandler.java

```diff
@@ -86,31 +85,40 @@ public void featureUsed(final DeprecatedFeatureUsage usage) {
             }
         }
         if (problemsService != null) {
-            InternalProblemReporter reporter = ((InternalProblems) problemsService).getInternalReporter();
-            Problem problem = reporter.create(new Action<InternalProblemSpec>() {
-                @Override
-                public void execute(InternalProblemSpec builder) {
-                    ProblemSpec problemSpec = builder
-                        // usage.getKind() could be be part of the problem ID, however it provides hints on the problem provenance which should be modeled differently, maybe as location data.
-                        .id("deprecated-feature-used", "Deprecated feature used", GradleCoreProblemGroup.deprecation())
-                        .contextualLabel(usage.getSummary())
-                        .documentedAt(usage.getDocumentationUrl());
-                    addPossibleLocation(diagnostics, problemSpec);
-                    problemSpec.severity(WARNING);
-                    if(usage.getAdvice() != null) {
-                        problemSpec.solution(usage.getAdvice());
-                    }
-                    if(usage.getContextualAdvice() != null) {
-                        problemSpec.solution(usage.getContextualAdvice());
-                    }
-                }
-            });
-            reporter.report(problem);
+            reportDeprecation(usage, diagnostics);
         }
         fireDeprecatedUsageBuildOperationProgress(usage, diagnostics);
     }
 
-    private static void addPossibleLocation(ProblemDiagnostics diagnostics, ProblemSpec genericDeprecation) {
+    private void reportDeprecation(final DeprecatedFeatureUsage usage, final ProblemDiagnostics diagnostics) {
+        InternalProblemReporter reporter = ((InternalProblems) problemsService).getInternalReporter();
+        Problem problem = reporter.create(new Action<InternalProblemSpec>() {
+            @Override
+            public void execute(InternalProblemSpec builder) {
+                InternalProblemSpec problemSpec = builder
+                    // usage.getKind() could be be part of the problem ID, however it provides hints on the problem provenance which should be modeled differently, maybe as location data.
+                    .id(usage.getProblemId(), usage.getSummary())
```

Comment:
❌ I don't think the summary is part of the ID. I think we need to introduce something new that is not the summary, since the summary is very context specific. We don't necessarily do this as part of this PR, though we should do it very soon.

Reply:
My assumption is that the summary is about as context specific as the id, since the id holds e.g. the class name and/or method name that is deprecated.
The elaborate description would be in "details".